core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java

```diff
@@ -502,7 +502,7 @@ public Address getAddress() {
       CompletableFuture<Map<Address, Response>> future = invokeRemotelyAsync(recipients, rpcCommand, mode,
             timeout, responseFilter, deliverOrder, anycast);
       try {
-         return future.get();
+         return CompletableFutures.await(future);
```

Comment:
Could we add some comments here why we can afford to wait endlessly here? What is providing the guarantees that we are sure that we'll never wait endlessly?